### PR TITLE
Add nsp security checker

### PR DESCRIPTION
**Pull request by pdehaan**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-    - "6.1.0"
+    - "6"
```
Comment (pdehaan) [inline .travis.yml]: Relaxed this pinned version, so it pulls latest 6.x builds.

File: package.json [hunk 2]
```diff
@@ -1,55 +1,60 @@
...
-  "homepage": "https://github.com/mozilla/blok#readme",
+  "dependencies": {
+  },
   "devDependencies": {
     "browserify": "13.0.1",
     "coveralls": "2.11.12",
     "envify": "3.4.1",
     "faucet": "0.0.1",
     "npm-watch": "0.1.5",
+    "nsp": "2.6.1",
```
Comment (pdehaan) [inline package.json]: Added this.
(Most of the other changes in this PR are me running `fixpack` which sorts and cleans up package.json files)

File: package.json [hunk 3]
```diff
@@ -1,55 +1,60 @@
...
+  "scripts": {
+    "build": "npm test && MODE=production npm run bundle && web-ext build -s src",
+    "build-dev": "npm test && MODE=dev npm run bundle && web-ext build -s src",
+    "bundle": "browserify -d -t envify src/js/background.js > src/js/background.bundle.js",
+    "coverage": "nyc npm test",
+    "coveralls": "nyc npm test && nyc report --reporter=text-lcov | coveralls",
+    "firefox": "npm run bundle && web-ext run -s src",
+    "lint": "standard && web-ext lint --self-hosted -s src",
+    "postlint": "nsp check -o summary",
+    "pretest": "npm run lint",
```
Comment (pdehaan) [inline package.json]: Added these two lines. Now running `npm run lint` will trigger nsp.
And running `npm test` will automatically run the `lint` script first.

File: .travis.yml (hunk 1 above)
Comment (groovecoder) [inline .travis.yml]: Is this standard practice for our node projects? I like pinning specific versions to keep control and remove the possibility of version incompat creating false-positive CI failures.

**Comment by groovecoder** [comment]
Out of curiosity, do you see this output during `lint`?
```
Error: Module did not self-register.
    at Error (native)
    at Object.Module._extensions..node (module.js:568:18)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/lcrouch/code/blok/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
{ Error: Cannot find module './build/default/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/lcrouch/code/blok/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3) code: 'MODULE_NOT_FOUND' }
{ Error: Cannot find module './build/Debug/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/lcrouch/code/blok/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3) code: 'MODULE_NOT_FOUND' }
```
It doesn't seem to break anything, but I'm not sure what it is. 😐

File: .travis.yml (hunk 1 above)
Comment (pdehaan) [inline .travis.yml]: I don't know about standards. We'd have to ask somebody like @jvehent maybe.
I just vaguely remember some recent Node security updates, and figured having a newer version isn't bad, but agree that there could be minor API differences in newer Nodes which break things.

**Comment by pdehaan** [comment]
> @groovecoder: _"Out of curiosity, do you see this output during `lint`?"_

I'll try a fresh clone and install. I sort of remember seeing that error on and off. I think it may be some odd optional dependency w/ DTrace on certain OSes.

**Comment by pdehaan** [comment]
> It doesn't seem to break anything, but I'm not sure what it is. 😐

@groovecoder I wasn't able to repro that specific error locally on this branch, but if you're seeing that, I'd probably say go scorched earth and do something radical like:

```sh
$ rm -rf node_modules && npm i
```

I'm wondering if maybe you installed some dependency using an older version of Node and then updated node and those gyp bindings maybe need to be updated.

**Comment by groovecoder** [comment]
Clearing `node_modules/` seems to have fixed it.

Let's go ahead and unpin from the minor release. But the first time it breaks something I'll probably grumble and re-pin us. 😉

**Pull request merged**